dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dataset/tab/DatasetFilesTabIT.java

```diff
@@ -0,0 +1,418 @@
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = LENIENT)
+@Transactional(ROLLBACK)
+public class DatasetFilesTabIT extends WebappArquillianDeployment {
+
+    DatasetFilesTabStubbed filesTab;
+
+    @Inject
+    DatasetRepository datasetRepo;
+    @Mock
+    FileDownloadHelper fileDownloadHelper;
+    @Inject
+    DataFileServiceBean datafileService;
+    @Inject
+    PermissionServiceBean permissionService;
+    @Mock
+    PermissionsWrapper permissionsWrapper;
+    @Inject
+    DataverseRequestServiceBean dvRequestService;
+    @Inject
+    DataverseSession session;
+    @Inject
+    GuestbookResponseServiceBean guestbookResponseService;
+    @Inject
+    EmbargoAccessService embargoAccess;
+    @Inject
+    SettingsServiceBean settingsService;
+    @Inject
+    EjbDataverseEngine commandEngine;
+    @Inject
+    ExternalToolServiceBean externalToolService;
+    @Inject
+    TermsOfUseFormMapper termsOfUseFormMapper;
+    @Mock
+    FileDownloadRequestHelper fileDownloadRequestHelper;
+    @Mock
+    GuestbookResponseDialog guestbookResponseDialog;
+    @Inject
+    ImageThumbConverter imageThumbConverter;
+    @Inject
+    FileMetadataService fileMetadataService;
+
+    DatasetFilesTabFacade datasetFilesTabFacade;
+    @Inject
+    ConfirmEmailServiceBean confirmEmailService;
+    @Inject
+    DatasetFieldValidationService fieldValidationService;
+    @Inject
+    DatasetDao datasetDao;
+    @Inject
+    DatasetVersionRepository datasetVersionRepository;
+    @Inject
+    AuthenticatedUserRepository userRepo;
+    @Inject
+    DataverseService dataverseService;
+
+    @Inject
+    LicenseRepository licenseRepo;
+    @Mock
+    UIComponent ui;
+    @Mock
+    Behavior behavior;
+
+    Dataset dataset;
+
+    @BeforeEach
+    public void setUp() {
+        this.dataset = this.datasetRepo.getById(52L);
+        this.datasetFilesTabFacade = new DatasetFilesTabFacade(
+                datasetVersionRepository, fileDownloadHelper, datasetDao);
+        this.filesTab = new DatasetFilesTabStubbed(fileDownloadHelper,
+                datafileService,
+                permissionService, permissionsWrapper,
+                dvRequestService, session,
+                guestbookResponseService, embargoAccess,
+                settingsService, commandEngine,
+                externalToolService, termsOfUseFormMapper,
+                fileDownloadRequestHelper, 
+                guestbookResponseDialog, imageThumbConverter,
+                fileMetadataService, datasetFilesTabFacade,
+                confirmEmailService, fieldValidationService);
+        this.filesTab.init(this.dataset.getLatestVersion());
+        this.filesTab.setFileLabelSearchTerm("");
+
+        AuthenticatedUser user = this.userRepo.getById(2L);
+        this.session.setUser(user);
+
+        when(this.permissionsWrapper.canViewUnpublishedDataset(any()))
+                .thenReturn(TRUE);
+    }
+
+    private static TermsOfUseForm licenseApacheForm() {
```

Comment:
Private methods should be located after the tests.

Reply:
fixed